src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRIORITY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_RATING;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.book.Priority;
+import seedu.address.model.book.Rating;
+import seedu.address.model.book.Status;
+
+/**
+ * Parses input arguments and creates a new {@code EditCommand} object
+ */
+public class EditCommandParser implements Parser<EditCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code EditCommand}
+     * and returns a {@code RateCommand} object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_RATING, PREFIX_PRIORITY,
+                PREFIX_STATUS);
+
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException ive) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+
+        int rating = Integer.valueOf(argMultimap.getValue(PREFIX_RATING).orElse("-1"));
+        String priority = argMultimap.getValue(PREFIX_PRIORITY).orElse("n");
+        String status = argMultimap.getValue(PREFIX_STATUS).orElse("u");
+        return new EditCommand(index, new Rating(rating), Priority.findPriority(priority),
```

Comment:
Check if the priority, status, and rating entered by the user is valid.
If the user input is invalid, throw a `ParseException`. You can refer to `ListCommandParser`.